File: app/src/main/java/com/routesme/taxi_screen/DetectInternetConnectionStatus/ConnectivityReceiver.java
package com.routesme.taxi_screen.DetectInternetConnectionStatus;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;
import android.widget.Toast;

import com.routesme.taxi_screen.Class.App;

public class ConnectivityReceiver extends BroadcastReceiver{
    private static final String TAG = "ConnectivityReceiver";

    public static ConnectivityReceiverListener connectivityReceiverListener;

    public ConnectivityReceiver() {
       // super();
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        //  Log.d(TAG, "From Connectivity Receiver ... Changed ... Internet Connection Status:  [ onReceive ] " );

        //Toast.makeText(context, "Action: " + intent.getAction(), Toast.LENGTH_SHORT).show();

        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        boolean isConnected = activeNetwork != null && activeNetwork.isConnectedOrConnecting();

        if (connectivityReceiverListener != null) {
            connectivityReceiverListener.onNetworkConnectionChanged(isConnected);
          //  Log.d(TAG, "From Connectivity Receiver ... Changed ... Internet Connection Status:  "+ isConnected);
        }
    }



    public static boolean isConnected() {
        ConnectivityManager cm = (ConnectivityManager) App.getInstance().getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
    }


    public interface ConnectivityReceiverListener {
        void onNetworkConnectionChanged(boolean isConnected);

    }


}

File: app/src/main/java/com/routesme/taxi/Class/ConnectivityReceiver.kt
package com.routesme.taxi.Class

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import java.util.*


class ConnectivityReceiver: BroadcastReceiver()  {

    class NetworkObservable private constructor() : Observable() {
        fun connectionChanged() {
            setChanged()
            notifyObservers()
        }

        companion object {
            var instance: NetworkObservable? = null
                get() {
                    if (field == null) {
                        field = NetworkObservable()
                    }
                    return field
                }
                private set
        }
    }

    fun getObservable(): NetworkObservable? {
        return NetworkObservable.instance
    }

    override fun onReceive(context: Context?, intent: Intent?) {

        Log.d("NetworkChangeReceiver","Connection status changed");
        //connectivityReceiverListener?.onNetworkConnectionChanged(getObservable()!!.connectionChanged();)
        getObservable()!!.connectionChanged();
    }
    /*private val connectivityStatus: Boolean
        get () {
            val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetwork = cm.activeNetworkInfo

            return activeNetwork != null && activeNetwork.isConnected
        }

    lateinit var context: Context
    override fun onReceive(context: Context, intent: Intent) {
        this.context = context
        connectivityReceiverListener?.onNetworkConnectionChanged(connectivityStatus)
        Log.d("Checking Network", connectivityStatus.toString())

    }

    interface ConnectivityReceiverListener {
        fun onNetworkConnectionChanged(isConnected: Boolean)
    }

    companion object {
        var connectivityReceiverListener: ConnectivityReceiverListener? = null
        val isConnected: Boolean
            get() {
                val cm = Objects.requireNonNull(instance)!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
                val activeNetwork = cm.activeNetworkInfo
                return activeNetwork != null && activeNetwork.isConnected       }
    }*/
}